docs/checkit.js

```diff
@@ -673,18 +685,24 @@ CheckIt.prototype.deleteCalendar = function() {
     var confirmation = confirm("Are you sure you want to delete your calendar?");
     if (confirmation) {
 
-        var currentCalendarId = this.store.getActive()
+        return this.store.getActive()
             .then(function(currentCalendarId) {
-                this.removeFromCalendarDropdown(currentCalendarId);
+                
+                this.removeFromCalendarDropdown(currentCalendarId, this.$calendarDropdown);
                 //delete the calendar and remove its active calendar status
-                this.store.removeById(currentCalendarId);
-                // To delete a calendar you have to be looking at it and if you
-                // are looking at it that means it is the currentActiveCalendar
-                this.store.removeActive();
-                //console.log("clearing the page");
-                //clear the page
-                this.clearPage(this.$calendarDiv);
-                this.showForm(this.$buildCalendarForm); 
+                return this.store.removeById(currentCalendarId)
```

Comment:
You can avoid have deep nesting by letting the fact that handlers that return Promises will "pass them through". You don't have to, just pointing it out. Note that you do need to put the catch handlers in the right place to catch the error at the right "level."

```
var checkit = this;
return checkit.store.getActive().then(function(currentCalendarId) {
  checkit.removeBlahBlah();
  return checkit.store.removeById(currentCalendarId).catch(function() {
    console.log('couldn't remove id from storage');
  });
}).then(function() {
  checkit.clearPage();
  checkit.showForm(..);
  return checkit.store.removeActive().catch(function() {console.log("couldn't remove active from store or whatever")});
}).catch(function() {console.log('couldn't delete');})
```